File: samples/composed/client/petstore/java/retrofit/src/test/java/io/swagger/client/api/PetApiTest.java
package io.swagger.client.api;

import io.swagger.client.ApiClient;
import io.swagger.client.model.Body1;
import io.swagger.client.model.Body2;
import io.swagger.client.model.InlineResponse200;
import io.swagger.client.model.InlineResponse2001;
import io.swagger.client.model.ModelApiResponse;
import io.swagger.client.model.Pet;
import org.junit.Before;
import org.junit.Test;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * API tests for PetApi
 */
public class PetApiTest {

    private PetApi api;

    @Before
    public void setup() {
        api = new ApiClient().createService(PetApi.class);
    }

    /**
     * Add a new parrow to the store
     *
     * 
     */
    @Test
    public void addParrotTest() {
        Body2 body = null;
        // InlineResponse2001 response = api.addParrot(body);

        // TODO: test validations
    }
    /**
     * Add a new pet to the store
     *
     * 
     */
    @Test
    public void addPetTest() {
        Pet body = null;
        // Void response = api.addPet(body);

        // TODO: test validations
    }
    /**
     * Deletes a pet
     *
     * 
     */
    @Test
    public void deletePetTest() {
        Long petId = null;
        String apiKey = null;
        // Void response = api.deletePet(petId, apiKey);

        // TODO: test validations
    }
    /**
     * Find pet by ID
     *
     * schedule pet feeding
     */
    @Test
    public void feedPetTest() {
        Pet body = null;
        String token = null;
        String petType = null;
        String status = null;
        Long petId = null;
        String sessionId = null;
        // Void response = api.feedPet(body, token, petType, status, petId, sessionId);

        // TODO: test validations
    }
    /**
     * Finds Pets by status
     *
     * Multiple status values can be provided with comma separated strings
     */
    @Test
    public void findPetsByStatusTest() {
        List<String> status = null;
        // List<Pet> response = api.findPetsByStatus(status);

        // TODO: test validations
    }
    /**
     * Finds Pets by tags
     *
     * Muliple tags can be provided with comma separated strings. Use\\ \\ tag1, tag2, tag3 for testing.
     */
    @Test
    public void findPetsByTagsTest() {
        List<String> tags = null;
        // List<Pet> response = api.findPetsByTags(tags);

        // TODO: test validations
    }
    /**
     * get Parrots
     *
     * 
     */
    @Test
    public void getParrotsTest() {
        // List<Object> response = api.getParrots();

        // TODO: test validations
    }
    /**
     * Find pet by ID
     *
     * Returns a single pet
     */
    @Test
    public void getPetByIdTest() {
        Long petId = null;
        // Pet response = api.getPetById(petId);

        // TODO: test validations
    }
    /**
     * update parrots
     *
     * 
     */
    @Test
    public void updateParrotsTest() {
        Body1 body = null;
        // InlineResponse200 response = api.updateParrots(body);

        // TODO: test validations
    }
    /**
     * Update an existing pet
     *
     * 
     */
    @Test
    public void updatePetTest() {
        Pet body = null;
        // Void response = api.updatePet(body);

        // TODO: test validations
    }
    /**
     * Updates a pet in the store with form data
     *
     * 
     */
    @Test
    public void updatePetWithFormTest() {
        Long petId = null;
        String name = null;
        String status = null;
        // Void response = api.updatePetWithForm(petId, name, status);

        // TODO: test validations
    }
    /**
     * uploads an image
     *
     * 
     */
    @Test
    public void uploadFileTest() {
        Long petId = null;
        Object body = null;
        // ModelApiResponse response = api.uploadFile(petId, body);

        // TODO: test validations
    }
}

File: samples/client/petstore/kotlin-retrofit2-client/src/test/kotlin/io/swagger/client/apis/PetApiTest.kt
package io.swagger.client.apis

import io.swagger.client.infrastructure.ApiClient
import io.swagger.client.models.ApiResponse
import io.swagger.client.models.Pet
import org.junit.Assert
import org.junit.Before
import org.junit.Test

/**
 * API tests for PetApi
 */
class PetApiTest {

    private lateinit var api: PetApi

    @Before
    fun setup() {
        api = ApiClient().createService(PetApi::class.java)
    }

    /**
     * Add a new pet to the store
     *
     * 
     */
    @Test
    fun addPetTest() {
        val body: Pet? = null
        // val response = api.addPet(body).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Deletes a pet
     *
     * 
     */
    @Test
    fun deletePetTest() {
        val petId: kotlin.Long? = null
        val apiKey: kotlin.String? = null
        // val response = api.deletePet(petId, apiKey).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Finds Pets by status
     *
     * Multiple status values can be provided with comma separated strings
     */
    @Test
    fun findPetsByStatusTest() {
        val status: kotlin.collections.List<kotlin.String>? = null
        // val response = api.findPetsByStatus(status).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Finds Pets by tags
     *
     * Multiple tags can be provided with comma separated strings. Use tag1, tag2, tag3 for testing.
     */
    @Test
    fun findPetsByTagsTest() {
        val tags: kotlin.collections.List<kotlin.String>? = null
        // val response = api.findPetsByTags(tags).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Find pet by ID
     *
     * Returns a single pet
     */
    @Test
    fun getPetByIdTest() {
        val petId: kotlin.Long? = null
        // val response = api.getPetById(petId).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Update an existing pet
     *
     * 
     */
    @Test
    fun updatePetTest() {
        val body: Pet? = null
        // val response = api.updatePet(body).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * Updates a pet in the store with form data
     *
     * 
     */
    @Test
    fun updatePetWithFormTest() {
        val petId: kotlin.Long? = null
        val name: kotlin.String? = null
        val status: kotlin.String? = null
        // val response = api.updatePetWithForm(petId, name, status).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }

    /**
     * uploads an image
     *
     * 
     */
    @Test
    fun uploadFileTest() {
        val petId: kotlin.Long? = null
        val additionalMetadata: kotlin.String? = null
        val file: java.io.File? = null
        // val response = api.uploadFile(petId, additionalMetadata, file).execute()
        // Assert.assertEquals(200, response.code())
        // TODO: test validations
    }
}
